### ISSUE TITLE
Early Stopping

### ISSUE THREAD

**Issue by Aditya715** [zalandoresearch/flair#864]
How can I implement early stopping. I'm working on NER and after a particular epoch the loss stops to decrease there I want to apply early stop. Kindly help me out.

**Comment by stefan-it** [comment]
Hi @Aditya715, by default `flair` uses an early stopping mechanism:

https://github.com/zalandoresearch/flair/blob/603970af3d393721a80426a69159ce0c61b3a468/flair/optim.py#L283

> Reduce learning rate and weight decay when a metric has stopped improving.

So when a learning rate of < 0.0001 is reached, training will be stopped :)

I hope this helps!

**Comment by alanakbik** [comment]
Perhaps we should add the stopping criterion as a parameter, i.e. let users specify at which point the learning rate becomes to small.

**Comment by alanakbik** [comment]
Closed by PR #871 - thanks again @stefan-it!
